### PR TITLE
os/se: RTL8730E ATF SE 1st release

### PR DESCRIPTION
**Pull request by ZhenBei-Sin**

### CODE REVIEW

File: os/net/lwip/src/core/netif.c
Comment (ZhenBei-Sin) [inline os/net/lwip/src/core/netif.c:1]: Hi @sunghan-chang, please solve the undefine reference for mbedtls_sha256.

File: build/configs/rtl8730e/hello/defconfig
Comment (ZhenBei-Sin) [inline build/configs/rtl8730e/hello/defconfig:1]: We are still in the progress of modifying the defconfig.

File: apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c [hunk 1]
```diff
@@ -51,6 +51,11 @@ static uint32_t SLOT1_START_ADDR = 0x1d000;
 static uint32_t SLOT0_START_ADDR = 0x24000;
 static uint32_t SLOT1_START_ADDR = 0x27000;
 #endif
+#ifdef CONFIG_AMEBASMART_TRUSTZONE
+/* ToDo: Temporary set need changes based flash addr */
+static uint32_t SLOT0_START_ADDR = 0x200000;
+static uint32_t SLOT1_START_ADDR = 0x203000;
```
Comment (sunghan-chang) [inline apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c:57]: Could you let me know what they are?

File: build/configs/rtl8730e/hello/defconfig
Comment (sunghan-chang) [reply build/configs/rtl8730e/hello/defconfig:1]: Please keep 644 for file permission.

File: build/configs/rtl8730e/rtl8730e_make_bin.sh [hunk 2]
```diff
@@ -71,7 +71,7 @@ FLOADER_PATH=${TOOL_PATH}/flashloader
 	# @#$(PREPENDTOOL) $(IMAGE_TARGET_FOLDER)/xip_image2.bin  __flash_text_start__  $(IMAGE_TARGET_FOLDER)/target_img2.map
 
 	# $(Q)cat $(IMAGE_TARGET_FOLDER)/ram_2.bin > $(IMAGE_TARGET_FOLDER)/ca32_image2_all.bin
-
+
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/rtl8730e_make_bin.sh:74]: Let's revert unnecessary change.

File: os/net/lwip/src/core/netif.c
Comment (sunghan-chang) [reply os/net/lwip/src/core/netif.c:1]: @ZhenBei-Sin I don't know what you mean. Currently there is no issue on build. How can I get the undefined reference you said?

File: os/net/lwip/src/core/netif.c [hunk 3]
```diff
@@ -1220,9 +1220,11 @@ err_t netif_gen_stable_private_id(struct netif *netif, s8_t addr_idx, ip6_addr_t
 		param.mac[i] = netif->hwaddr[i];
 	}
 	// ToDo : mbedTLS is in userspace. so mbedtls_sha256 can't be called.
+/*
```
Comment (sunghan-chang) [inline os/net/lwip/src/core/netif.c:1223]: I think this is related with what you said above. Even it is, you should not block tizenrt common code like this. Please revert this change.

File: apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c (hunk 1 above)
Comment (ZhenBei-Sin) [reply apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c:57]: This is the temporary address for Secure Storage, so that the test case can be passed.

File: os/board/rtl8730e/src/Makefile
Comment (ZhenBei-Sin) [inline os/board/rtl8730e/src/Makefile:1]: Hi @sunghan-chang ,  Currently there is no issue on the build due to RTK Make.defs including the mbedtls file.
RTK kernel code should not include mbedtls that is why we do the action on removal, it can it will cause the undefined reference of mbedtls_sha256.

File: build/configs/rtl8730e/hello/defconfig [hunk 4]
```diff
@@ -273,27 +280,42 @@ CONFIG_AMEBASMART_FLASH_BASE=0x0
...
 CONFIG_FLASH_MINOR=0
-CONFIG_FLASH_PART_SIZE="48,76,4,3260,3260,512,1024,8,"
-CONFIG_FLASH_PART_TYPE="none,none,none,kernel,kernel,smartfs,ftl,bootparam,"
-CONFIG_FLASH_PART_NAME="bl1,reserved,system_data,kernel,kernel,userfs,reserved,bootparam,"
+#CONFIG_FLASH_PART_SIZE="48,76,4,3260,3260,512,1024,8,"
+#CONFIG_FLASH_PART_TYPE="none,none,none,kernel,kernel,smartfs,ftl,bootparam,"
+#CONFIG_FLASH_PART_NAME="bl1,reserved,system_data,kernel,kernel,userfs,reserved,bootparam,"
+CONFIG_FLASH_PART_SIZE="128,1792,6264,8,"
+CONFIG_FLASH_PART_TYPE="none,kernel,none,bootparam,"
+CONFIG_FLASH_PART_NAME="bl1,kernel,reserved,bootparam,"
```
Comment (sunghan-chang) [inline build/configs/rtl8730e/hello/defconfig:296]: 1. Could you let me know why you need to modify like this?
2. Where is the secure storage partition?
3. Let's keep `userfs, reserved, bootparam,` at the end with size `512, 1024, 8,`.

File: os/board/rtl8730e/src/Makefile
Comment (sunghan-chang) [reply os/board/rtl8730e/src/Makefile:1]: It's the TizenRT lwip code. I am sorry but I don't know why you want to modify that. And I don't know the issue exactly.
Is it
1. when you remove mbedtls code in lwip, you meet the undefined reference error?
2. when you DON'T remove mbedtls code in lwip, you do?

File: build/configs/rtl8730e/hello/defconfig (hunk 4 above)
Comment (ZhenBei-Sin) [reply build/configs/rtl8730e/hello/defconfig:296]: Hi @sunghan-chang, this is the temporary flash layout design. 
We haven't decided on the final flash layout.
We will take note of your point 3 for the final flash layout.

Thank You.

File: os/board/rtl8730e/src/Makefile
Comment (ZhenBei-Sin) [reply os/board/rtl8730e/src/Makefile:1]: 1. when you remove mbedtls code in lwip, you meet the undefined reference error?
Ans: When we remove the include mbedtls at our RTK kernel makefile "os/board/rtl8730e/src/Makefile", then will face undefined reference of mbedtls_sha256 "os/net/lwip/src/core/netif.c". 
It seems like TizenRT lwip code relies on RTK kernel makefile.
 
2. when you DON'T remove mbedtls code in lwip, you do?
Ans: Yes, I will revert the changes on TizenRT lwip /* for the mbedtls_sha256. 
But Samsung side needs to take care of the build error that throws from the TizenRT lwip code.

For this issue, we had feedback to @kishore-sn & @abhishek-samsung, and Samsung side work in progress on it.

Comment (abhishek-samsung) [reply os/board/rtl8730e/src/Makefile:1]: hello could you please remove Realtek related mbedtls code from the Makefile and enable the config https://github.com/Samsung/TizenRT/blob/932c9e3b7e43949fe4d6aab8186e3405ee166537/build/configs/rtl8730e/hello/defconfig#L820 

This adds mbedtls from TizenRT side. 
https://github.com/Samsung/TizenRT/blob/932c9e3b7e43949fe4d6aab8186e3405ee166537/external/mbedtls/Make.defs#L53-L55

File: build/configs/rtl8730e/hello/defconfig [hunk 5]
```diff
@@ -222,19 +221,27 @@ CONFIG_ARCH_INTERRUPTSTACK=4096
...
+CONFIG_AMEBASMART_BOOT_DRAM=y
+# CONFIG_AMEBASMART_BOOT_NOR is not set
+# CONFIG_AMEBASMART_BOOT_SRAM is not set
+
 #
 # Boot Memory Configuration
 #
+CONFIG_RAM_KREGIONx_START="0x60400000,"
+CONFIG_RAM_KREGIONx_SIZE="4194304,"
+##### Need to crosscheck
```
Comment (abhishek-samsung) [inline build/configs/rtl8730e/hello/defconfig:236]: Looks like the defconfig has been reverted back. Could you please rebase to newer code base? Kconfig changes were made and the defconfig was modified in the PR https://github.com/Samsung/TizenRT/pull/5844

File: os/board/rtl8730e/src/Makefile
Comment (ZhenBei-Sin) [reply os/board/rtl8730e/src/Makefile:1]: Hi @abhishek-samsung , noted with thanks.

File: os/se/ameba/rtl_security_api_wrapper_tz.c [hunk 6]
```diff
@@ -101,6 +101,25 @@ static void ns_printf(const char *input)
...
+
+
+#ifdef CONFIG_AMEBASMART_TRUSTZONE
+void *rtl_set_ns_func(void)
+{
+	/* No need for RTL8730E */
+	return NULL;
+}
+
+void up_allocate_secure_context(size_t size)
```
Comment (ZhenBei-Sin) [inline os/se/ameba/rtl_security_api_wrapper_tz.c:113]: Hi @sunghan-chang,

Currently, we have added up_allocate_secure_context() and up_free_secure_context() to allow SE to run without error.
In 8730E we are not using ARMV8M_TRUSTZONE, thus these 2 API is not present. However, RTL SE APIs called it, should we add a define switch to all the caller locations, or Samsung will help to add these 2 dummy APIs in TizenRT Kernel? If Samsung will help to add dummy APIs, these 2 dummy APIs in RTL SE can be removed.

File: build/configs/rtl8730e/hello/defconfig
Comment (sunghan-chang) [reply build/configs/rtl8730e/hello/defconfig:1]: This file is still changed to 755. Please keep original permission 644.

File: build/configs/rtl8730e/hello/defconfig (hunk 5 above)
Comment (ZhenBei-Sin) [reply build/configs/rtl8730e/hello/defconfig]: Hi @abhishek-samsung, we had done updated the defconfig.
Thank You.

**Review by sunghan-chang — APPROVED** [review]
